api/lib/opentelemetry/trace/link.rb

```diff
@@ -32,8 +32,7 @@ class Link
       #   this link. Attributes will be frozen during Link initialization.
       # @return [Link]
       def initialize(span_context, attributes = nil)
-        raise ArgumentError unless span_context.instance_of?(SpanContext)
-        raise ArgumentError unless Internal.valid_attributes?(attributes)
+        attributes = nil unless Internal.valid_attributes?(attributes)
```

Comment:
This check has the same overhead as the raising version. Does it buy us much? Should we keep the "check with default" or handle the failing cases later, during serialization?